src/fontra/client/web-components/modal-dialog.js

```diff
@@ -236,26 +234,33 @@ export class DialogOverlay extends SimpleElement {
   }
 
   show() {
-    this._savedActiveElement = document.activeElement;
-    this.style.display = "grid";
-    this.dialogBox.focus();
+    this._tabbed = false;
+    this.dialogElement.showModal();
   }
 
   hide() {
-    this.style.display = "none";
-    this._savedActiveElement?.focus();
+    this.dialogElement.close();
+    this._tabbed = false;
   }
 
   _handleKeyDown(event) {
-    if (event.key == "Enter") {
-      if (!this.defaultButton?.classList.contains("disabled")) {
-        this.defaultButton?.click();
+    if (event.key === "Enter") {
+      // console.log(document.activeElement);
+      // unfortunately document.activeElement is always the whole dialog,
```

Comment:
See findNestedActiveElement() function for an example of how to find the active element within a shadow-root active element:

https://github.com/googlefonts/fontra/blob/c80e68bf4bbb803aa73f8bbc0bbbe5a9746d34b0/src/fontra/client/core/utils.js#L316-L322